Add notebook re: cartopy for rotated lon/lat grid plots

douglatornell opened the pull request:
Notebook rendered at https://nbviewer.jupyter.org/github/UBC-MOAD/PythonNotes/blob/cartopy/CartopyRotatedGrid.ipynb

douglatornell committed:
Add cartopy pkg to conda environment

douglatornell committed:
Add cmocean pkg to conda environment

douglatornell committed:
Add notebook re: cartopy for lon/lat grid plots

douglatornell committed:
Add linebreak to improve code reaability

SusanEAllen commented:
Very nice.  This notebook will be very useful for our team switching to cartopy.  Two niggles:

1. Not sure about 29o East.  I'd say 29o counter-clockwise.

2. Is the final figure actually square?  It looks stretched in the along strait direction to me.

douglatornell committed:
Address Susan's comments in PR #1 review

* Drop confusing attempt to describe rotation direction of SalishSeaCast
  NEMO grid and just let the description of it being aligned with the
  SoG alnog-strait axis do the job
* Drop set_aspect() calls for lon/lat grid plots; unnecessary because
  plotting on lons/lats implicitly makes the aspect ratio correct

Also make some miscellaneous other improvements to the code:

* Change to set axes background land colour in add_subplot() calls,
  either directly or via subplot_kw arg in subplots()
* Improve explanation of why we can't use xarray.plot() for lon/lat
  plots re: lons/lats of SalishSeaCast NEMO grid are 2d arrays that
  make no sense as coordinates
* Change to use fig.tight_layout() instead of plt.tight_layout()
  because the former is less magical
* Add fig.canvas.draw() before fig.tight_layout() because it's needed
  by cartopy
